data/zilliz/v0/createCollection.go

```diff
@@ -0,0 +1,110 @@
+package zilliz
+
+import (
+	"fmt"
+
+	"github.com/instill-ai/component/base"
+	"google.golang.org/protobuf/types/known/structpb"
+)
+
+const (
+	createCollectionPath = "/v2/vectordb/collections/create"
+)
+
+type CreateCollectionOutput struct {
+	Status string `json:"status"`
+}
+
+type CreateCollectionInput struct {
+	CollectionName   string         `json:"collection-name"`
+	Dimension        int            `json:"dimension"`
+	MetricType       string         `json:"metric-type"`
+	IDType           string         `json:"id-type"`
+	AutoID           bool           `json:"auto-id"`
+	PrimaryFieldName string         `json:"primary-field-name"`
+	VectorFieldName  string         `json:"vector-field-name"`
+	Schema           map[string]any `json:"schema"`
+	IndexParams      map[string]any `json:"index-params"`
+	Params           map[string]any `json:"params"`
+}
+
+type CreateCollectionReq struct {
+	CollectionNameReq   string         `json:"collectionName"`
+	DimensionReq        int            `json:"dimension"`
+	MetricTypeReq       string         `json:"metricType"`
+	IDTypeReq           string         `json:"idType"`
+	AutoIDReq           bool           `json:"autoID"`
+	PrimaryFieldNameReq string         `json:"primaryFieldName"`
+	VectorFieldNameReq  string         `json:"vectorFieldName"`
+	SchemaReq           map[string]any `json:"schema"`
+	IndexParamsReq      map[string]any `json:"indexParams"`
+	ParamsReq           map[string]any `json:"params"`
+}
+
+type CreateCollectionResp struct {
+	Code    int            `json:"code"`
+	Message string         `json:"message"`
+	Data    map[string]any `json:"data"`
+}
+
+func (e *execution) createCollection(in *structpb.Struct) (*structpb.Struct, error) {
+	var inputStruct CreateCollectionInput
+	err := base.ConvertFromStructpb(in, &inputStruct)
+	if err != nil {
+		return nil, err
+	}
+
+	resp := CreateCollectionResp{}
+
+	reqParams := CreateCollectionReq{
+		CollectionNameReq: inputStruct.CollectionName,
+		MetricTypeReq:     inputStruct.MetricType,
+		AutoIDReq:         inputStruct.AutoID,
+		DimensionReq:      inputStruct.Dimension,
+	}
+	if inputStruct.IDType != "" {
+		reqParams.IDTypeReq = inputStruct.IDType
+	}
+	if inputStruct.PrimaryFieldName != "" {
+		reqParams.PrimaryFieldNameReq = inputStruct.PrimaryFieldName
+	}
+	if inputStruct.VectorFieldName != "" {
+		reqParams.VectorFieldNameReq = inputStruct.VectorFieldName
+	}
+	if inputStruct.Schema != nil {
+		reqParams.SchemaReq = inputStruct.Schema
+	}
+	if inputStruct.IndexParams != nil {
+		reqParams.IndexParamsReq = inputStruct.IndexParams
+	}
+	if inputStruct.Params != nil {
+		reqParams.ParamsReq = inputStruct.Params
+	}
+
+	req := e.client.R().SetBody(reqParams).SetResult(&resp)
+
+	res, err := req.Post(createCollectionPath)
+
+	if err != nil {
+		return nil, err
+	}
+
+	if res.StatusCode() != 200 {
```

Comment:
Please help me confirm the status code when the action is post

Reply:
According to its documentation, code will always be 0 for success request, otherwise it indicates errors